packages/firestore/src/core/transaction.ts

```diff
@@ -132,8 +133,8 @@ export class Transaction {
       throw fail('Document in a transaction was a ' + doc.constructor.name);
     }
 
-    const existingVersion = this.readVersions.get(doc.key);
-    if (existingVersion !== null) {
+    const existingVersion = this.readVersions.get(doc.key.toString());
+    if (existingVersion) {
```

Comment:
Why did you drop the null check? Is `get` guaranteed to return an object or null?

Reply:
Maps return `undefined` for non-existing values and I prefer `if(value)` over `if(value !=== undefined)`.